pkg/trace/writer/stats_sync.go

```diff
@@ -0,0 +1,208 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+package writer
+
+import (
+	"math"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/trace/config"
+	"github.com/DataDog/datadog-agent/pkg/trace/info"
+	"github.com/DataDog/datadog-agent/pkg/trace/logutil"
+	"github.com/DataDog/datadog-agent/pkg/trace/metrics"
+	"github.com/DataDog/datadog-agent/pkg/trace/metrics/timing"
+	"github.com/DataDog/datadog-agent/pkg/trace/stats"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const maxPendingStatsPayloads = 10
+
+// StatsSyncWriter ingests stats buckets and flushes them to the API.
+type StatsSyncWriter struct {
```

Comment:
There seems to be a lot of duplication here. Can't we have the original trace/stats writers and have them simply take a flag which alters their behaviour? I mean 90% of this code is the same as the original one. That way we could also get rid of the interfaces.

Reply:
@gbbr I've gone through and merged the logic from the two files. I think you're right, it is neater.